src/components/Missions/MissionsPage.js

```diff
@@ -1,16 +1,32 @@
 const MissionsPage = (prop) => {
-  const { name, description } = prop;
+  const {
+    joinMissionHandler, id, name, description, reserved, leaveMissionHandler,
+  } = prop;
+
+  let button;
+  if (!reserved) {
+    button = <button type="button" className="mission-button-join" onClick={() => joinMissionHandler(id)}>Join Mission</button>;
+  } else {
+    button = <button type="button" className="mission-button-badge" onClick={() => leaveMissionHandler(id)}>Leave Mission</button>;
+  }
+
+  let badge;
+  if (!reserved) {
+    badge = <span className="mission-button">Not a Member</span>;
+  } else {
+    badge = <span className="join-badge ">Active Member</span>;
+  }
```

Comment:
In the future, you can create functions that render your component like:

> function RenderButton( reserved){
>  if (!reserved) {
>        return <button type="button" className="mission-button-join" onClick={() => joinMissionHandler(id)}>Join 
                   Mission</button>;
>   } else {
>    return   <button type="button" className="mission-button-badge" onClick={() => 
                                             leaveMissionHandler(id)}>Leave Mission</button>;
>   }
> } 

if you do this, you can call this function in the future you can turn this function into a component and reuse it